homeassistant/components/incomfort/binary_sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Support for an Intergas boiler via an InComfort/InTouch Lan2RF gateway."""
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up an InComfort/InTouch binary_sensor device."""
+    client = hass.data[DOMAIN]['client']
+    heater = hass.data[DOMAIN]['heater']
+
+    async_add_entities([
+        IncomfortBurning(client, heater),
+        IncomfortPumping(client, heater),
+        IncomfortTapping(client, heater),
+        IncomfortFailed(client, heater)
+    ])
+
+
+class IncomfortBinarySensor(BinarySensorDevice):
+    """Representation of an InComfort/InTouch binary_sensor device."""
+
+    def __init__(self, client, boiler):
+        """Initialize the binary sensor."""
+        self._client = client
+        self._boiler = boiler
+
+        self._name = None
+        self._is_on_key = None
+        self._other_key = None
+
+    async def async_added_to_hass(self):
+        """Set up a listener when this entity is added to HA."""
+        async_dispatcher_connect(self.hass, DOMAIN, self._refresh)
+
+    @callback
+    def _refresh(self):
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return the status of the sensor."""
+        return self._boiler.status[self._is_on_key]
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        if self._other_key is None:
+            return None
+        value = self._boiler.status[self._other_key] if self.is_on else None
+        return {self._other_key: value}
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False as this device should never be polled."""
+        return False
+
+
+class IncomfortBurning(IncomfortBinarySensor):
+    """Representation of an InComfort Burning sensor."""
+
+    def __init__(self, client, boiler):
+        """Initialize the binary sensor."""
+        super().__init__(client, boiler)
+
+        self._name = 'Burning'
+        self._is_on_key = 'is_burning'
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        if self._boiler.status['is_pumping']:
+            value = self._boiler.status['heater_temp']
```

Comment:
Yes, for water_heater, but no. The boiler has no stored water.  It has _two_ temps:
 - heater_temp for the circulating volume (CV, for central heating) that leaves the boiler and returns to be re-heated, and 
 - tap_temp for DHW (potable domestic host water for drinking, showers, etc.) that leaves and never returns

The question was how to show both temps when required...

Reply:
So are they already reported for water_heater entity?

Reply:
Yes.

Reply:
IF it comes to that, it may make sense to just remove the water_heater entity?  Or temp only, and no extra state_attributes:
```python
        if self._heater.is_tapping:
            return self._heater.tap_temp
        elif self._heater.is_pumping:
            return self._heater.heater_temp
        return max(self._heater.heater_temp, self._heater.tap_temp)
```

Reply:
You know the integration best. We shouldn't duplicate information but think and decide where it fits best. :+1:

Reply:
OK, I'll think about it, but my instincts are to:
 - remove most (all?) of the device_state_attributes from the water_heater
 - swap two of the binary sensors (pumping, tapping) to temperature sensors, with the on/off state as an attribute
 - remove the pumping on/off sensor in favour of CV pressure

Reply:
I would suggest keeping binary values as binary sensors, and temp values as sensor entities, unless the temp values are naturally part of water heater. Is there some identifying device name for the sensors that we can show? Then it will still be clear what sensors belong to what device.

In the future you can add config entry and device info to connect the sensors to the same device.

Reply:
It is all one piece of hardware - a combi/system boiler!

There is an intimate connection between some values: say `tap_temp`, and `is_tapping` (tap temp will rise when is_tapping is True, and fall to room temperature otherwise).  That's why I wanted one of the two to be a sensor, and the other to be a device_state_attribute of that sensor device.

What is the difference between is_tapping as a device_state_attribute of tap_temp, and as a separate binary sensor?  For me, the big difference is that, by default, HA will show a graph.

> In the future you can add config entry and device info to connect the sensors to the same device.

Anything I can read about now?

Reply:
Home Assistant represents devices as entities of different domains. As far as possible device values and properties should be represented as the different domain entity states.

Generally if a value can represent the device property on its own it should be returned as an entity state. Only values of device properties that can't relay the device property on its own, should be combined with another value and be reported as state attributes.

Take eg the light case. The light can be on/off. From this value we know if the light is shining or not. We put this value as the light state. The light can also have a brightness. Unless we encode more information into the brightness value, we can't know by the brightness value alone, if the light is shining or not. We put the brightness as a state attribute.

Back to our case. We know from `is_tapping` if the water is being tapped or not. We put it as a binary sensor state. We also know from `tap_temp` what temperature the tap water has. We put it as a sensor state.

https://developers.home-assistant.io/docs/en/config_entries_index.html

Reply:
I have had a big think about this, and am happy with it the way it is now - Water_heater, sensors x3, and a fault_state sensor.